Automattic/wp-calypso
# Stats: Add support Jetpack Stats banner

Related to #78653

## Proposed Changes

The PR proposes to add 'Do you love Jetpack Stats' notice, which would be shown when the site is Jetpack and the site doesn't have valid purchase (Paid). When the notice is dismissed, the notice would be snoozed for 30 days.

## Testing Instructions


* Set up Store sandbox `pejTkB-FS-p2`
* Sandbox your Jetpack
* Sandbox `public-api.wordpress.com` with D115186-code
* Build the branch for Odyssey
  * STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev
* Build Jetpack if necessary
  * jetpack build plugins/jetpack
* Open `/wp-admin/admin.php?page=stats&flags=stats/paid-stats`
* Ensure you see the notice
* Purchase Jetpack Stats `https://wordpress.com/checkout/{site-slug}/jetpack_stats_monthly`
* Ensure the notice is gone

Note: please test on Calypso Stats for a Jetpack site too

<img width="826" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/e15a7b5b-3481-4a84-85d9-a4cd4d2cfb84">

<img width="870" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1425433/512c963b-a74c-417d-aa66-2522c2fef66b">



## Pre-merge Checklist
